### PR TITLE
Add the ability to have a favorites courses

### LINKED ISSUE #103: modify db
now every registered user has his own favourite courses arranged in a table in the db, also with relation to courses

### PR DESCRIPTION
**Pull request by mosesnagar**
resolve #103 
Each user have the functionality to add favorites courses. 
`favorites` table added to handle `many-to-many` relationship.

### CODE REVIEW

File: tests/test_favorites_courses.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+from tutor import db
+from tutor.models.users import Users
+from tutor.models.course import Course
+
+
+def test_fav(client):
+    # create new user
+    u = Users(username='pypy', email='email@email.com', password='pass')
+    u.save()
```
Comment (golan13) [inline tests/test_favorites_courses.py:9]: not really an issue, but you can use the function `create_test_user` from `user_functions.py`

File: tests/test_favorites_courses.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    # check that user exists and has no favorites
+    user = Users.query.filter_by(username='pypy').first()
```
Comment (golan13) [inline tests/test_favorites_courses.py:12]: Same here, can use `search_user_name`

File: tests/test_favorites_courses.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+    # check the course added to user favorites
+    same_user = Users.query.filter_by(username='pypy').first()
+    assert c == same_user.favorites[0]
+
+    # clean up db
+    same_user.removeAllFavorites()
+    db.session.delete(c)
+    db.session.delete(same_user)
+    db.session.commit()
```
Comment (golan13) [inline tests/test_favorites_courses.py:27]: there is a `cleanup` function in `contest.py` that cleans all the db after running tests. You can remove all favourites there

**Review by golan13 — APPROVED** [review]

**Review by Michal-Shamir — APPROVED** [review]
Looks good

**Review by yuvalmargules — APPROVED** [review]

**Review by LiorRey — APPROVED** [review]
LGTM

**Review by golan13 — APPROVED** [review]

**Pull request merged**